### DISCUSSION TITLE
Duplicate binding YAML file

### DISCUSSION THREAD

**Discussion by ofirshe** [zephyrproject-rtos/zephyr#69425] [Q&A]
Hello,

I'm currently engaged in a Zephyr-based project involving out-of-tree development. I've successfully created my own driver for the MAXIM MAX31790 PWM fan controller. However, upon the latest synchronization from the Zephyr main branch, I've noticed that a driver for the same device has been developed and merged into the main branch. Consequently, I'm encountering a compilation error:

`devicetree error: both C:/Users/ofirs/bsp/dts/bindings\fan\maxim,max31790.yaml and C:/Users/ofirs/bsp/zephyr/dts/bindings\pwm\maxim,max31790.yaml have 'compatible: maxim,max31790' and 'on-bus: i2c'`

Is there a way to 'force' the operating system to include my custom driver in the build and disregard the upstreamed one, without altering the name of the binding file? I'm keen on retaining my driver.

Appreciate your assistance,
Ofir

**Comment by henrikbrixandersen** [comment]
I'm afraid not. You'll have to come up with a custom "compatible" for your out-of-tree driver if the upstream binding isn't sufficient for you.
